x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartDataFrameAnalyticsAction.java

```diff
@@ -674,22 +669,8 @@ protected void nodeOperation(AllocatedPersistentTask task, TaskParams params, Pe
             );
 
             // Get stats to initialize in memory stats tracking
-            ActionListener<Boolean> templateCheckListener = ActionListener.wrap(
-                ok -> executeAsyncWithOrigin(client, ML_ORIGIN, GetDataFrameAnalyticsStatsAction.INSTANCE,
-                    new GetDataFrameAnalyticsStatsAction.Request(params.getId()), statsListener),
-                error -> {
-                    Throwable cause = ExceptionsHelper.unwrapCause(error);
-                    logger.error(
-                        new ParameterizedMessage(
-                            "[{}] failed to create internal index template [{}]",
-                            params.getId(),
-                            inferenceIndexTemplate.getTemplateName()),
-                        cause);
-                    dfaTask.setFailed(error);
-                }
-            );
-
-            MlIndexAndAlias.installIndexTemplateIfRequired(clusterState, client, inferenceIndexTemplate, templateCheckListener);
```

Comment:
I may misunderstand this but:

- analytics initially started running on a cluster before we had inference model system index
- rolling upgrade starts
- analytics gets assigned to a new node (master is still on old version)

Does the system index automatically get applied when we call `PUT <doc>` against the inference index?

Reply:
Good point.  It doesn't work for the case where the master is still on the old version.